docs/dev/table/connectors/kafka.md

```diff
@@ -249,6 +249,15 @@ Besides enabling Flink's checkpointing, you can also choose three different mode
 
 Please refer to [Kafka documentation]({% link dev/connectors/kafka.md %}#kafka-producers-and-fault-tolerance) for more caveats about delivery guarantees.
 
+### Per-partition-watermark Source
+
+Flink supports to emit per-partition-watermark for Kafka. Using this feature, watermarks are generated inside the Kafka consumer. The per-partition-watermark are merged in
+the same way as watermarks are merged on stream shuffles. The output watermark of the source is determined by the minimum watermark among the partitions it reads. Considering a watermark assigner
+advance the watermark according to the event-time on the records. If some partitions in the topics are idle, the watermark assigner will not advance. You can alleviate this problem by
+setting appropriate [idelness timeouts]({{ site.baseurl }}/dev/event_timestamps_watermarks.html#dealing-with-idle-sources).
```

Comment:
This documentation makes no sense for Table/SQL users. We don't have a possibility to set this via options, right?

Reply:
We should actually introduce an option for it.

Reply:
Could you give a bad case about this feature? With this feature, we can get more fine-grained watermark. Only some partition are idle, the watermark assigner doesn't advance itself. However, we can set idle timeout to solve this problem.

Reply:
Maybe there is some misunderstanding. The link that you put in the docs helps DataStream API users:
```
WatermarkStrategy
        .<Tuple2<Long, String>>forBoundedOutOfOrderness(Duration.ofSeconds(20))
        .withIdleness(Duration.ofMinutes(1));
```

But it should be updated to help Table/SQL API users.

I found `org.apache.flink.table.api.config.ExecutionConfigOptions#TABLE_EXEC_SOURCE_IDLE_TIMEOUT` we should link to this instead if it works for the Kafka connector.

Reply:
You are right. I modify the link: [table.exec.source.idle-timeout](https://ci.apache.org/projects/flink/flink-docs-master/dev/table/config.html#table-exec-source-idle-timeout) and [Kafka watermark strategies](https://ci.apache.org/projects/flink/flink-docs-master/dev/event_timestamps_watermarks.html#watermark-strategies-and-the-kafka-connector).